src/detect-engine-helper.c

```diff
@@ -57,7 +57,7 @@ InspectionBuffer *DetectHelperGetData(struct DetectEngineThreadCtx_ *det_ctx,
             return NULL;
 
         InspectionBufferSetup(det_ctx, list_id, buffer, b, b_len);
-        InspectionBufferApplyTransforms(buffer, transforms);
+        InspectionBufferApplyTransforms(det_ctx, buffer, transforms);
```

Comment:
Would you not want to generalize usage of `DetectHelperGetData` so that changing the prototype of `InspectionBufferApplyTransforms` affects less files ?

Reply:
That's a good suggestion, but it will affect many source modules.

Can we consider that an optimization to follow this PR?

Reply:
This PR affects many source modules :-p

It would affect less and be more focused if this code factorization happened first ;-)